filer/utils/compatibility.py

```diff
@@ -76,3 +76,19 @@ def get_delete_permission(opts):
 except ImportError:
     # python < 2.7
     from django.utils.importlib import import_module  # flake8: noqa
+
+
+try:
+    from urllib import urlencode  # flake8: noqa
+except:
+    # python 3
+    from urllib.parse import urlencode  # flake8: noqa
+
+
+try:
+    from PIL import Image as PILImage
```

Comment:
django officially dropped support for PIL and replaced it with Pillow. Might be worth also checking for that